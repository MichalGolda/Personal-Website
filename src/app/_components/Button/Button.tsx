export type ButtonProps = {
  className?: string;
};

const Button: React.FC<React.PropsWithChildren<ButtonProps>> = ({
  className = "",
  children,
  ...props
}) => {
  return (
    <button
      className={`py-4 px-8 font-bold transition-all duration-500 ${className}`}
      {...props}
    >
      {children}
    </button>
  );
};

export default Button;
