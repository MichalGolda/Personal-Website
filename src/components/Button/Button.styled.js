import styled from "styled-components";
import { variant } from "styled-system";

export const StyledButton = styled.button`
  padding: 16px 32px;
  border: none;
  column-gap: 32px;
  display: inline-flex;
  font-weight: 700;
  cursor: pointer;
  width: fit-content;
  background-color: transparent;
  font-size: 16px;
  transition: ${({ theme }) => theme.transition};

  ${variant({
    variants: {
      primary: {
        color: "white",
        backgroundColor: "secondary",
      },
      outlined: {
        color: "secondary",
        borderColor: "secondary",
        borderWidth: "2px",
        borderStyle: "solid",
        "&:hover": {
          backgroundColor: "secondary",
          color: "white",
        },
      },
      outlinedLoading: {
        pointerEvents: "none",
        color: "tertiary",
        borderColor: "tertiary",
        borderWidth: "2px",
        borderStyle: "solid",
      },
    },
  })};
`;
