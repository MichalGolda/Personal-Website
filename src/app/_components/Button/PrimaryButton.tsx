import Button from "./Button";
import type { ButtonProps } from "./Button";

const PrimaryButton: React.FC<React.PropsWithChildren<ButtonProps>> = ({
  className = "",
  children,
  ...props
}) => {
  return (
    <Button
      className={`bg-primary text-white hover:bg-primary/90 active:bg-primary/80 border border-transparent border-solid ${className}`}
      {...props}
    >
      {children}
    </Button>
  );
};

export default PrimaryButton;
