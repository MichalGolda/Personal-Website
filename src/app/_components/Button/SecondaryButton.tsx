import Button from "./Button";
import type { ButtonProps } from "./Button";

const SecondaryButton: React.FC<React.PropsWithChildren<ButtonProps>> = ({
  className = "",
  children,
  ...props
}) => {
  return (
    <Button
      className={`bg-secondary text-white hover:bg-secondary/90 active:bg-secondary/80 border border-transparent ${className}`}
      {...props}
    >
      {children}
    </Button>
  );
};

export default SecondaryButton;
