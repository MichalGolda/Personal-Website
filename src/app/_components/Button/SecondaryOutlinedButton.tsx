import Button from "./Button";
import type { ButtonProps } from "./Button";

const SecondaryOutlinedButton: React.FC<
  React.PropsWithChildren<ButtonProps>
> = ({ className = "", children, ...props }) => {
  return (
    <Button
      className={`bg-transparent border-secondary border border-solid hover:bg-secondary active:bg-secondary/80 hover:text-white text-secondary ${className}`}
      {...props}
    >
      {children}
    </Button>
  );
};

export default SecondaryOutlinedButton;
