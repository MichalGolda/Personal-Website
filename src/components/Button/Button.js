import { StyledButton } from "./Button.styled";

export default function Button({ children, ...props }) {
  return <StyledButton {...props}>{children}</StyledButton>;
}
