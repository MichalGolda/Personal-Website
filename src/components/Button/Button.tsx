import React from "react";
import { StyledButton } from "./Button.styled";

export type ButtonProps = {
  type?: "button" | "submit" | "reset";
  variant: "primary" | "outlined" | "outlinedLoading";
};

const Button: React.FC<React.PropsWithChildren<ButtonProps>> = ({
  children,
  ...props
}) => <StyledButton {...props}>{children}</StyledButton>;

export default Button;
