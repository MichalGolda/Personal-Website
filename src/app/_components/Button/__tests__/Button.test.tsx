import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import Button from "../Button"

describe("Button component", () => {
    it("renders without errors", () => {
        render(<Button type="button">Button test</Button>)
    
        const button = screen.getByRole("button")

        expect(button).toBeInTheDocument()
        expect(button).toHaveAttribute("type", "button")
    })

    it("renders unchanged", () => {
        const { container } = render(<Button type="button">Button test</Button>)
        
        expect(container).toMatchSnapshot()
    })
})
