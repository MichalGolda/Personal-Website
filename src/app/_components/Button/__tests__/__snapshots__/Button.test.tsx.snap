// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders unchanged 1`] = `
<div>
  <button
    class="py-4 px-8 font-bold text-sm md:text-base transition-all duration-500 disabled:bg-lightGrey disabled:text-body "
    type="button"
  >
    Button test
  </button>
</div>
`;
